astrapy/db.py

```diff
@@ -1987,7 +1989,11 @@ def create_collection(
         # Build the final json payload
         jsondata = {
             k: v
-            for k, v in {"name": collection_name, "options": options}.items()
+            for k, v in {
+                "name": collection_name,
+                "options": options,
+                "indexing": indexing,
```

Comment:
Careful. "indexing" is not a new top-level param as it seemed initially.
It was later [confirmed](https://datastax.slack.com/archives/C04RPUB5JRM/p1706134743975189?thread_ts=1706101652.447869&cid=C04RPUB5JRM) that it is inside the "options" dict.

So...
1. not urgent to sugar the indexing into a friendlier way since one can take advantage of the feature right away
2. in case of separate indexing parameters to the method, similarly to 'dimension' and 'metric' it is necessary to check that it is not passed twice (once within the options and once as separate argument to create_collection)

And for what is worth, if sugar has to be, I would favor two parameters to the effect of allow and deny (and a check that no both are nonempty at the same time)